functions/tests.py

```
@@ -0,0 +1,161 @@
+def mock_az_secrets(monkeypatch):
+    store = {}
+
+    class MockSecretClient:
+        def set_secret(self, name, value):
+            store[name] = value
+
+        def get_secret(self, name):
+            try:
+                return SimpleNamespace(value=store[name])
+            except KeyError:
+                raise azure.core.exceptions.ResourceNotFoundError()
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)